modules/authglobus/lib/Auth/Source/Globus.php

```diff
@@ -0,0 +1,273 @@
+     * Constructor for this authentication source.
+     *
+     * @param array $info  Information about this authentication source.
+     * @param array $config  Configuration.
+     */
+    public function __construct($info, $config) {
+        // Call the parent constructor first, as required by the interface
+        parent::__construct($info, $config);
+        
+        $this->key = $config['key'];
+        $this->secret = $config['secret'];
+        $this->scope = $config['scope'];
+        $this->response_type = $config['response_type'];
+        $this->redirect_uri= $config['redirect_uri'];
+        $this->curl = curl_init();
+    }
+
+    /**
+     * Flattens an associative array, derived from LinkedIn module's flatten
+     *
+     * @param array $array
+     * @param string $prefix
+     *
+     * @return array the array with the new concatenated keys
+     */
+    protected function flatten($array, $prefix = '') {
+        $newArr = array();
+
+        foreach ($array as $key => $value) {
+            if (is_array($value)) {
+                $newArr = $newArr + $this->flatten($value, $prefix . $key . '.');
+            } else {
+                $newArr[$prefix . $key] = $value;
+            }
+        }
+
+        return $newArr;
+    }
+
+    /**
+    * Accepts a field containing a full name, attempts to extract a full name
+    *
+    * @param string $name
+    *
+    * @return an array containing first name and last name
+    */
+    protected function getFullName($name) {
+        $trimmedName = trim($name);
+
+        if ($trimmedName && (strpos($trimmedName, ',') !== null || strpos($trimmedName, '\s') !== null)) {
+            $delimiter = strpos($name, ',') !== null ? ',' : '\s';
+            $nameParts = explode($delimiter, $trimmedName, 2);
+            $nameParts = array_map('trim', $nameParts);
+
+            $firstName;
+            $lastName;
+
+            if ($delimiter === '\s') {
+                $firstName = $nameParts[0];
+                $lastName = $nameParts[1];
+            } else {
+                $firstName = $nameParts[1];
+                $lastName = $nameParts[0];
+            }
+
+            return array(
+                'first_name' => $firstName,
+                'last_name' => $lastName
+            );
+        }
+
+        return array(
+            'first_name' => 'UNKNOWN',
+            'last_name' => 'UNKNOWN'
+        );
+    }
+
+    /**
+    * Wrapper for Curl Requests
+    *
+    * @param string path
+    * @param string signature
+    * @param string token
+    * @param string qs
+    * @param array contents
+    * @param string method
+    *
+    * @return The response from the endpoint where we made the request.
+    */
+    protected function doCurlRequest(
+        $path,
+        $signature = null,
+        $token = null,
+        $qs = null,
+        $contents = null,
+        $method = 'GET'
+    ) {
+        curl_setopt($this->curl, CURLOPT_USERAGENT, 'SSPHP Globus'); //umad Globus
```

Comment:
something seems odd about how we are doing this.

We are initializing curl once, but then everytime we make a curl request, we reset the options again?

Reply:
I was picturing a scenario in which more of them might change at some point depending on calls we're making. Agreed that we can move some though